### PR TITLE
feat: improve code

### LINKED ISSUE #4: False positive if query has leading whitespace
In our code we have SQL in constants, and for readability they start with an empty line:

```
err := s.db.QueryRow(someSQL).Scan(&foo)

const someSQL = `
  SELECT foo 
  FROM bar
  WHERE ...query continues
`
```

On this input we get an `execinquery` error:

```
It's better to use Execute method instead of QueryRow method to execute `` query
```

Suggestion: `TrimLeft` the query string.

### LINKED ISSUE #5: incorrect match
The following snippet is incorrectly recognized by the linter:
```go
package query

import "context"

const selectCodes = `-- name: SelectCodeForUpdate :one
SELECT code, max_uses, uses, ttl, enabled, created_at, updated_at
FROM codes
AS OF SYSTEM TIME follower_read_timestamp()
WHERE code = $1
`

func (q *Queries) SelectRegisterCodeFollowerRead(ctx context.Context, code string) (Code, error) {
	row := q.db.QueryRow(ctx, selectCodes , code)
	var i RegisterBonusCode
	err := row.Scan(
		&i.Code,
		&i.MaxUses,
		&i.Uses,
		&i.Ttl,
		&i.Enabled,
		&i.CreatedAt,
		&i.UpdatedAt,
	)
	return i, err
}
```

@lufeee

### PR DESCRIPTION
**Pull request by ldez**
- Support `var`, `const`, and concatenation.
- Trim comments.

Fixes #5
Fixes #4

### CODE REVIEW

**Review by leonid-shevtsov — APPROVED** [review]

**Comment by ldez** [comment]
@lufeee can you review/merge the PR? I want to create a bug-fix release of golangci-lint.

**Comment by favonia** [comment]
I made a further PR into this PR as ldez/execinquery#1 in an attempt to provide shorter and more accurate suggestions. This would close #11.
- Before: ``It's better to use Execute method instead of QueryRowContext method to execute `UPDATE` query``
- After: ``Use ExecContext instead of QueryRowContext to execute `UPDATE` query``

File: execinquery.go [hunk 1]
```diff
@@ -49,53 +44,80 @@ func run(pass *analysis.Pass) (interface{}, error) {
...
+			if query == "" {
 				return
 			}
 
-			s = strings.ToTitle(strings.SplitN(s, " ", 2)[0])
+			query = strings.TrimSpace(cleanValue(query))
+			parts := strings.SplitN(query, " ", 2)
+			cmd := strings.ToTitle(parts[0])
+
+			if strings.HasPrefix(strings.ToUpper(cmd), "SELECT") {
```
[the remark this answers is not in the document]
Comment (lufeee) [reply execinquery.go:67]: MySQL can make query such as `SeleCT * from users` , so I use `strings.ToUpper`

[the remark this answers is not in the document]
Comment (ldez) [reply execinquery.go]: https://go.dev/play/p/v3NyibUZvo3

[the remark this answers is not in the document]
Comment (favonia) [reply execinquery.go]: One could argue whether we should actually use `ToUpper`, but it makes no differences for valid SQL queries.

File: execinquery.go [hunk 2]
```diff
@@ -49,53 +44,80 @@ func run(pass *analysis.Pass) (interface{}, error) {
...
-			if strings.HasPrefix(strings.ToLower(s), "select") {
+			query := getQueryString(n.Args[i])
+			if query == "" {
 				return
 			}
 
-			s = strings.ToTitle(strings.SplitN(s, " ", 2)[0])
+			query = strings.TrimSpace(cleanValue(query))
+			parts := strings.SplitN(query, " ", 2)
+			cmd := strings.ToTitle(parts[0])
```
Comment (favonia) [inline execinquery.go:65]:
```suggestion
			cmd := strings.ToUpper(parts[0])
```
Final nitpicking---let's just use the obvious function to kill any possible misunderstanding. :stuck_out_tongue:

**Review by favonia — APPROVED** [review]
This would be an amazing enhancement to the current linter!

**Comment by lufeee** [comment]
This is fine, but there is a /* as a comment in the SQL. The current code does not support that, and the case where the -- comment is two columns.
`-- foobar
-- foobar
DELETE * FROM test WHERE test=?
`

**Comment by ldez** [comment]
I will fix the multiple `--` but I think `\*` can be handled in another PR.

**Comment by lufeee** [comment]
OK

**Comment by ldez** [comment]
> Would this be easy to implement using this regex function for /*?
> `regexp.MustCompile("/*[^\n]+*/").ReplaceAllString(v, "")`

The expression must be different because `[^\n]` will not work in this case, I will try something.

**Review by favonia — COMMENTED** [review]
Maybe we could use global variables to hold these regexps? I don't have a strong opinion.

**Comment by ldez** [comment]
I dislike global variables, then I prefer creating a struct.

File: execinquery.go [hunk 3]
```diff
@@ -29,7 +29,7 @@ type linter struct {
 
 func newLinter() *linter {
 	return &linter{
-		commentExp:          regexp.MustCompile("-- [^\n]+\n"),
+		commentExp:          regexp.MustCompile(`-[^\n]*\n`),
```
Comment (favonia) [inline execinquery.go:32]:
```suggestion
		commentExp:          regexp.MustCompile(`--[^\n]*\n`),
```
I think you accidentally deleted the second dash. :laughing:

**Comment by lufeee** [comment]
Its good 
thanks guys!

**Pull request merged**